instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/metrics/db/DbConnectionPoolMetrics.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.metrics.db;
+
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.DoubleHistogram;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.MeterBuilder;
+import io.opentelemetry.api.metrics.ObservableLongUpDownCounter;
+import io.opentelemetry.instrumentation.api.internal.EmbeddedInstrumentationProperties;
+import java.util.function.LongSupplier;
+
+/**
+ * A helper class that models the <a
+ * href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/database-metrics.md#connection-pools">database
+ * client connection pool metrics semantic conventions</a>.
+ */
+public final class DbConnectionPoolMetrics {
+
+  static final AttributeKey<String> POOL_NAME = stringKey("pool.name");
+  static final AttributeKey<String> CONNECTION_STATE = stringKey("state");
+
+  static final class ConnectionState {
+
+    static final String IDLE = "idle";
+    static final String USED = "used";
+  }
+
+  public static DbConnectionPoolMetrics create(
+      OpenTelemetry openTelemetry, String instrumentationName, String poolName) {
+
+    MeterBuilder meterBuilder = openTelemetry.getMeterProvider().meterBuilder(instrumentationName);
+    String version = EmbeddedInstrumentationProperties.findVersion(instrumentationName);
+    if (version != null) {
```

Comment:
The one impl has a `@Nullable` field so the guard _could_ be omitted. Maybe this is general good practice, but it's still more code to read.

Reply:
The API method does not have `@Nullable` though -- until now we've strictly followed the API guidelines even when we knew that the implementation handles nulls well.

Reply:
Yeah, that's completely sane.